packages/astro/src/types/public/integrations.ts

```diff
@@ -115,6 +115,22 @@ export interface AstroAdapter {
 	 * If the adapter is not able to handle certain configurations, Astro will throw an error.
 	 */
 	supportedAstroFeatures: AstroAdapterFeatureMap;
+	/**
+	 * Configuration for Astro's client-side code.
+	 */
+	client?: {
+		/**
+		 * Headers to inject into Astro's internal fetch calls (Actions, View Transitions, Server Islands, Prefetch).
+		 * Can be an object of headers or a function that returns headers.
+		 */
+		internalFetchHeaders?: Record<string, string> | (() => Record<string, string>);
+		/**
+		 * Query parameters to append to all asset URLs (images, stylesheets, scripts, etc.).
+		 * Should be a string in the format "key=value&other=value".
+		 * Useful for adapters that need to track deployment versions or other metadata.
+		 */
+		assetQueryParams?: string;
```

Comment:
Wouldn't it be better if this was a Map, plain object, or even `URLSearchParams`?

Reply:
So you can be sure it's the right format, rather than appending an arbitrary string to the URLs